### PR TITLE
Reuse builders for output examples

### LINKED ISSUE #1058: [Task]: Update examples to use Builders from Outputs
### Task description

Update examples to use Alias/Nft/Foundry Builders from Outputs as in https://github.com/iotaledger/wallet.rs/pull/1158/files

### Requirements

N/A

### Acceptance criteria

N/A

### Creation checklist

- [ ] I have assigned this task to the correct people
- [X] I have added the most appropriate labels
- [X] I have linked the correct milestone and/or project

### PR DESCRIPTION
**Pull request by pvdrz**

### CODE REVIEW

File: examples/outputs/alias.rs [hunk 1]
```diff
@@ -52,21 +52,19 @@ async fn main() -> Result<()> {
...
-            .add_unlock_condition(UnlockCondition::StateControllerAddress(
-                StateControllerAddressUnlockCondition::new(address),
-            ))
-            .add_unlock_condition(UnlockCondition::GovernorAddress(GovernorAddressUnlockCondition::new(
-                address,
-            )))
-            .finish_output()?,
-    ];
+    let alias_output_builder = AliasOutputBuilder::new_with_amount(1_000_000, AliasId::null())?
+        .with_foundry_counter(0)
```
Comment (Thoralf-M) [inline examples/outputs/alias.rs]: This shouldn't be necessary, since 0 is default
```suggestion
```

File: examples/outputs/alias.rs [hunk 2]
```diff
@@ -52,21 +52,18 @@ async fn main() -> Result<()> {
...
+        .add_feature(Feature::Metadata(MetadataFeature::new(vec![1, 2, 3])?))
+        .add_immutable_feature(Feature::Issuer(IssuerFeature::new(address)))
+        .add_unlock_condition(UnlockCondition::StateControllerAddress(
+            StateControllerAddressUnlockCondition::new(address),
+        ))
+        .add_unlock_condition(UnlockCondition::GovernorAddress(GovernorAddressUnlockCondition::new(
+            address,
+        )));
+
+    let outputs = vec![alias_output_builder.clone().with_state_index(0).finish_output()?];
```
Comment (Thoralf-M) [inline examples/outputs/alias.rs]:
```suggestion
    let outputs = vec![alias_output_builder.clone().finish_output()?];
```

File: examples/outputs/all_automatic_input_selection.rs [hunk 3]
```diff
@@ -61,23 +61,29 @@ async fn main() -> Result<()> {
...
+    // address of the owner of the NFT
+    let nft_output_builder = NftOutputBuilder::new_with_amount(1_000_000, NftId::null())?
+        .add_unlock_condition(UnlockCondition::Address(AddressUnlockCondition::new(address)));
+
     let outputs = vec![
-        AliasOutputBuilder::new_with_amount(2_000_000, AliasId::null())?
+        alias_output_builder
+            .clone()
             .with_state_index(0)
             .with_foundry_counter(0)
```
Comment (Thoralf-M) [inline examples/outputs/all_automatic_input_selection.rs]:
```suggestion
```

File: examples/outputs/foundry.rs [hunk 4]
```diff
@@ -60,19 +60,23 @@ async fn main() -> Result<()> {
...
+        .add_unlock_condition(UnlockCondition::GovernorAddress(GovernorAddressUnlockCondition::new(
+            address,
+        )));
+
     let outputs = vec![
-        AliasOutputBuilder::new_with_amount(2_000_000, AliasId::null())?
+        alias_output_builder
+            .clone()
             .with_state_index(0)
             .with_foundry_counter(0)
```
Comment (Thoralf-M) [inline examples/outputs/foundry.rs]:
```suggestion
```

**Review by Thoralf-M — APPROVED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**